avocado/core/multiplexer.py

```diff
@@ -377,28 +377,67 @@ def iteritems(self):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
+def _report_already_parsed(self, *args, **kwargs):
+    """
+    Raises exception describing that `self.data` alteration is restricted
+    """
+    raise RuntimeError("Altering of `self.data` is restricted. %s %s"
```

Comment:
Well the problem is that we need to pickle the object, so I have to create static function in a global namespace. The static method, nor function generator is pickable (unless I create a specific function, which would still be a problem should one replace this function with another).

That is the reason why I used generic function and I don't distinguish between the functions. At the same time I wanted to list all the arguments available as it might help identifying what causes it (along with the traceback).